pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/LeafStageTransferableBlockOperator.java

```diff
@@ -113,6 +113,9 @@ public String toExplainString() {
 
   @Override
   protected TransferableBlock getNextBlock() {
+    if (_isFinished) {
```

Comment:
Why do we need this? We shouldn't need this unless the upstream operator breaks the contract of not calling next block when EOS is already returned

Reply:
correct. i think there's a bug in the behavior of Window & Aggregate Operator that causes this. previously this was handled b/c MailboxReceivedOperator will return multiple EOS.

I can correct that behavior but decided to move that to a separate PR.

Reply:
fixed in https://github.com/apache/pinot/pull/11970.